### PR TITLE
Update CDS Services to 1.29.0

### PR DESCRIPTION
**Pull request by beckermarc**
- Use H2 v2 with new h2 dialect provided by CDS Compiler
- Use new name java instead of old java-cf for build tasks
- Update dependencies

### CODE REVIEW

File: .cdsrc.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
         "target": ".",
         "tasks": [
             {
-                "for": "java-cf"
+                "for": "java"
```
Comment (BraunMatthias) [inline .cdsrc.json:6]: are you sure about this? We still seem to run classic mtx in the bookshop

Comment (beckermarc) [reply .cdsrc.json:6]: I think this change is independent of streamlined or classic MTX, but we can also keep it back if you like.

Comment (beckermarc) [reply .cdsrc.json:6]: I'd suggest to go for it. @mofterdinger also proposed a similar change to the projects that we generate in the cds-services archetype.

Comment (BraunMatthias) [reply .cdsrc.json:6]: indeed` java-cf` and `java `point to the same implementation. `java-cf` should be replaced.

**Review by BraunMatthias — APPROVED** [review]

**Pull request merged**